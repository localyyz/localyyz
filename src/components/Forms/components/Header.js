import React from "react";
import { View, StyleSheet, Text } from "react-native";

// custom
import { Styles } from "localyyz/constants";

export default class Header extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text numberOfLines={1} style={[styles.label, this.props.labelStyle]}>
          {this.props.children}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {},

  label: {
    ...Styles.Text,
    ...Styles.SmallText
  }
});
