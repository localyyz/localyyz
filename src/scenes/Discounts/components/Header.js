import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";

// third party
import { inject, observer } from "mobx-react/native";

// custom
import { Colours, Sizes, Styles } from "localyyz/constants";
import DealType from "./DealType";
import FeaturedDealsCarousel from "./FeaturedDealsCarousel";

@observer
export default class Header extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <FeaturedDealsCarousel />
        <DealType />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colours.Foreground
  }
});
