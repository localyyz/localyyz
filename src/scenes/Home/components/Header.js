import React from "react";
import { Animated, View, StyleSheet, StatusBar } from "react-native";

// third party
import { withNavigation } from "react-navigation";
import { observer, inject } from "mobx-react/native";
import * as Animatable from "react-native-animatable";

// custom
import { Styles, Colours, Sizes } from "localyyz/constants";
import { navLogo } from "localyyz/assets";

// constants
const START_FADE_DISTANCE = 0;
const END_FADE_DISTANCE = Sizes.Height / 5;
const FADE_DISTANCE = [START_FADE_DISTANCE, END_FADE_DISTANCE];

@inject(stores => ({
  scrollAnimate: stores.homeStore.scrollAnimate,
  setHeaderHeight: height => {
    stores.homeStore.headerHeight = height;
  }
}))
@observer
export class Header extends React.Component {
  onLayout = ({ nativeEvent: { layout: { height } } }) => {
    this.props.setHeaderHeight(height);
  };

  render() {
    const bkgColorAnimate = this.props.scrollAnimate.interpolate({
      inputRange: FADE_DISTANCE,
      outputRange: ["white", Colours.Transparent],
      extrapolate: "clamp"
    });
    const bkgHeightAnimate = this.props.scrollAnimate.interpolate({
      inputRange: FADE_DISTANCE,
      outputRange: [Sizes.InnerFrame * 2.5, 0],
      extrapolate: "clamp"
    });
    const tintColorAnimate = {
      tintColor: this.props.scrollAnimate.interpolate({
        inputRange: FADE_DISTANCE,
        outputRange: [Colours.Text, Colours.Transparent],
        extrapolate: "clamp"
      })
    };
    return (
      <Animated.View
        onLayout={this.onLayout}
        style={[styles.container, { backgroundColor: bkgColorAnimate }]}>
        <Animated.View
          style={[
            styles.search,
            { backgroundColor: bkgColorAnimate, height: bkgHeightAnimate }
          ]}>
          <View style={styles.idleContainer}>
            <Animatable.View animation="fadeInDown" duration={800} delay={200}>
              <Animated.Image
                style={[styles.logo, tintColorAnimate]}
                source={navLogo}/>
            </Animatable.View>
          </View>
        </Animated.View>
      </Animated.View>
    );
  }
}

export default withNavigation(Header);

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0
  },

  search: {
    marginTop: Sizes.ScreenTop,
    overflow: "hidden"
  },

  idleContainer: {
    ...Styles.Horizontal,
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  },

  logo: {
    height: 13,
    width: 90,
    tintColor: Colours.AlternateText
  }
});
