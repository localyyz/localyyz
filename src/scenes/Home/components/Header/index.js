import React from "react";
import {
  Animated,
  View,
  StyleSheet,
  Text,
  StatusBar,
  TouchableWithoutFeedback
} from "react-native";

// third party
import { withNavigation } from "react-navigation";
import { inject } from "mobx-react/native";
import * as Animatable from "react-native-animatable";

// custom
import { Styles, Colours, Sizes } from "localyyz/constants";
import { navLogo } from "localyyz/assets";

// constants
const START_FADE_DISTANCE = 0;
const END_FADE_DISTANCE = Sizes.Height / 5;
const FADE_DISTANCE = [START_FADE_DISTANCE, END_FADE_DISTANCE];

@inject(stores => ({
  // layout
  scrollAnimate: stores.homeStore.scrollAnimate,
  homeStore: stores.homeStore
}))
class Header extends React.Component {
  constructor(props) {
    super(props);

    // refs
    this.inputRef = React.createRef();

    // bindings
    this.renderIdle = this.renderIdle.bind(this);
  }

  renderIdle() {
    return (
      <View style={styles.idleContainer}>
        <Animatable.View animation="fadeInDown" duration={800} delay={200}>
          <Animated.Image
            style={[
              styles.logo,
              {
                tintColor: this.props.scrollAnimate.interpolate({
                  inputRange: FADE_DISTANCE,
                  outputRange: [Colours.Text, Colours.AlternateText],
                  extrapolate: "clamp"
                })
              }
            ]}
            source={navLogo}/>
        </Animatable.View>
      </View>
    );
  }

  render() {
    return (
      <Animated.View
        style={[
          styles.container,
          {
            backgroundColor: this.props.scrollAnimate.interpolate({
              inputRange: FADE_DISTANCE,
              outputRange: [Colours.Transparent, Colours.MenuBackground],
              extrapolate: "clamp"
            })
          }
        ]}>
        <StatusBar barStyle="light-content" />
        <Animated.View
          style={[
            styles.search,
            {
              backgroundColor: this.props.scrollAnimate.interpolate({
                inputRange: FADE_DISTANCE,
                outputRange: [Colours.WhiteTransparent, Colours.Highlight],
                extrapolate: "clamp"
              })
            }
          ]}>
          {this.renderIdle()}
        </Animated.View>
      </Animated.View>
    );
  }
}

export default withNavigation(Header);

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: Colours.MenuBackground
  },

  search: {
    margin: Sizes.InnerFrame,
    marginTop: Sizes.ScreenTop,
    height: Sizes.InnerFrame * 2.5,
    paddingHorizontal: Sizes.InnerFrame,
    borderRadius: Sizes.InnerFrame,
    overflow: "hidden"
  },

  idleContainer: {
    ...Styles.Horizontal,
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  },

  searchLabelContainer: {
    position: "absolute",
    left: Sizes.InnerFrame / 2,
    top: 2,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center"
  },

  searchLabel: {
    ...Styles.Text,
    ...Styles.Subdued,
    ...Styles.TinyText
  },

  logo: {
    height: 13,
    width: 90,
    tintColor: Colours.AlternateText
  }
});
