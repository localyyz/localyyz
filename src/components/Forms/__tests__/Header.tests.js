import React from "react";
import Renderer from "react-test-renderer";

// local
import Header from "../components/Header";

// constants
const NAME = "Header Test - Form";
const PROPS = {
  labelStyle: { color: "green" },
  children: "sample children"
};

const Component = Header;

describe(NAME, () => {
  it(`${NAME}: should render properly`, () => {
    let c = Renderer.create(<Component {...PROPS} />).getInstance();
    expect(c).toBeDefined();
  });

  it(`${NAME}: should render to snapshot`, () => {
    expect(
      Renderer.create(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });
});
