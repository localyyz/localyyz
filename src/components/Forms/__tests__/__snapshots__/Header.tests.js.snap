// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test - Form Header Test - Form: should render to snapshot 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "rgb(30, 30, 30)",
          "fontSize": 14,
          "fontWeight": "300",
        },
        Object {
          "color": "green",
        },
      ]
    }
  >
    sample children
  </Text>
</View>
`;
